src/System.Buffers/tests/ArrayPool/UnitTests.cs

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Buffers;
+using Xunit;
+
+namespace System.Buffers.ArrayPool.Tests
+{
+    public partial class ArrayPoolUnitTests
+    {
+        /*
+            NOTE - due to test parallelism and sharing, use an instance pool for testing unless necessary
+        */ 
+        [Fact]
+        public static void SharedInstanceCreatesAnInstanceOnFirstCall()
+        {
+            Assert.True(ArrayPool<byte>.Shared != null);
+        }
+
+        [Fact]
+        public static void SharedInstanceOnlyCreatesOneInstanceOfOneTypep()
+        {
+            ArrayPool<byte> instance = ArrayPool<byte>.Shared;
+            Assert.True(instance == ArrayPool<byte>.Shared);
+        }
+
+        [Fact]
+        public static void CreateWillCreateMultipleInstancesOfTheSameType()
+        {
+            Assert.True(ArrayPool<byte>.Create() != ArrayPool<byte>.Create());
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(-1)]
+        public static void CreatingAPoolWithInvalidArrayCountThrows(int length)
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() => ArrayPool<byte>.Create(numberOfArrays: length));
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(-1)]
+        [InlineData(15)]
+        public static void CreatingAPoolWithInvalidMaximumArraySizeThrows(int length)
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() => ArrayPool<byte>.Create(maxArrayLength: length));
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(-1)]
+        public static void RentingWithInvalidLengthThrows(int length)
+        {
+            ArrayPool<byte> pool = ArrayPool<byte>.Create();
+            Assert.Throws<ArgumentOutOfRangeException>(() => pool.Rent(length));
+        }
+
+        [Fact]
+        public static void RentingAValidSizeArraySucceeds()
+        {
+            Assert.True(ArrayPool<byte>.Create().Rent(100) != null);
+        }
+
+        [Fact]
+        public static void RentingMultipleArraysGivesBackDifferentInstances()
+        {
+            ArrayPool<byte> instance = ArrayPool<byte>.Create(numberOfArrays: 2);
+            Assert.True(instance.Rent(100) != instance.Rent(100));
+        }
+
+        [Fact]
+        public static void RentingMoreArraysThanSpecifiedInCreateWillStillSucceed()
+        {
+            ArrayPool<byte> instance = ArrayPool<byte>.Create(numberOfArrays: 1);
+            Assert.True(instance.Rent(100) != null);
+            Assert.True(instance.Rent(100) != null);
```

Comment:
Fixed and added a new test to validate that we can get back a bigger array than was requested